src/azul/terraform.py

```diff
@@ -721,6 +722,29 @@ def tf_config(self, app_name):
                 key.replace('.', '_'): value
                 for key, value in bucket_notifications.items()
             }
+            # To prevent a race condition by Terraform, we make the bucket
+            # notifications depend on the related aws_lambda_permission and
+            # aws_lambda_function resources.
+            for notification in resources['aws_s3_bucket_notification'].values():
+                lambda_arns = [
+                    lambda_function['lambda_function_arn']
```

Comment:
If the notification resource already refers to an attribute of the lambda function resource, there is no point in adding an explicit dependency of the notification on the lambda function.